spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterTablePartitionFields.java

```diff
@@ -301,17 +284,33 @@ public void testReplacePartition() {
     sql("ALTER TABLE %s ADD PARTITION FIELD days(ts)", tableName);
     table.refresh();
     PartitionSpec expected =
-        PartitionSpec.builderFor(table.schema()).withSpecId(1).day("ts").build();
+        PartitionSpecParser.fromJson(
```

Comment:
I had to switch to JSON so that I can correctly validate field IDs used in the spec, which is not accessible in a programatic way in this module.